### PR TITLE
[docs] upgrade notes for styleguide

### PR DESCRIPTION
**Pull request by TomasGottvald**
| Q             | A
| ------------- | ---
|Description, reason for the PR| adds upgrage notes for styleguide introduced in #1463
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 1]
```diff
@@ -543,7 +543,7 @@ There you can find links to upgrade notes for other versions too.
             +         return $this->getContainer();
             +       }
             ```
-    - to achieve the goal you should find and replace all occurrences of accessing class directly from container, e.g. `$this->getContainer()->get(FooBar::class)` and define it as a class property with an inject annotation instead 
+    - to achieve the goal you should find and replace all occurrences of accessing class directly from container, e.g. `$this->getContainer()->get(FooBar::class)` and define it as a class property with an inject annotation instead
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: This is auto change done by my editor. And it is only format changes. Should I remove this format changes?

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 2]
```diff
@@ -639,6 +639,85 @@ There you can find links to upgrade notes for other versions too.
         ```
         - for more information you can see the [PR](https://github.com/shopsys/shopsys/pull/1461)
 
+- add Basic styleguide ([#1463](https://github.com/shopsys/shopsys/pull/1463))
+    - update md files according to PR
+        - [docs/frontend/basic-design-styleguide.md](https://github.com/shopsys/shopsys/blob/master/docs/frontend/basic-design-styleguide.md)
+        - [docs/frontend/navigation.yml](https://github.com/shopsys/shopsys/blob/master/docs/frontend/navigation.yml)
+        - [docs/index.md](https://github.com/shopsys/shopsys/blob/master/docs/index.md)
+        - [open-source-license-acknowledgements-and-third-party-copyrights.md](https://github.com/shopsys/shopsys/tree/master/open-source-license-acknowledgements-and-third-party-copyrights.md)
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: Removed.

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 3]
```diff
@@ -491,43 +491,43 @@ There you can find links to upgrade notes for other versions too.
...
                     /**
             -        * @var \Shopsys\FrameworkBundle\Component\Domain\Domain|null
             +        * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
             +        * @inject
                      */
             -       private $domain;
             +       protected $domain;
-    
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: This is auto change done by my editor. And it is only format changes. Should I remove this format changes?

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 4]
```diff
@@ -639,6 +639,85 @@ There you can find links to upgrade notes for other versions too.
...
+        +     <exclude name="/"/>
+        + </fileset>
+        ```
+
+    - update config file [services.yml](https://github.com/shopsys/shopsys/tree/master/packages/framework/src/Resources/config/services.yml)
+        ```diff
+        - ['admin/', 'common/', 'custom_admin/', 'frontend/']
+        + ['admin/', 'common/', 'custom_admin/', 'frontend/', 'styleguide/']
+        ```
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: Do you mean, that changes in framework should not be described here? Bcs. we changes framework files

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 5]
```diff
@@ -639,6 +639,85 @@ There you can find links to upgrade notes for other versions too.
...
+    - update config file [.eslintignore](https://github.com/shopsys/shopsys/tree/master/project-base/.eslintignore)
+        ```diff
+          /src/Shopsys/ShopBundle/Resources/scripts/frontend/plugins
+        + /src/Shopsys/ShopBundle/Resources/scripts/styleguide
+        ```
+
+    - add route and contorller according to PR
+        update `project-base/app/config/routing_dev.yml`
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: linked

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 6]
```diff
@@ -639,6 +639,85 @@ There you can find links to upgrade notes for other versions too.
...
+        + use Shopsys\ShopBundle\Controller\Front\FrontBaseController;
+        +
+        + class StyleguideController extends FrontBaseController
+        + {
+        +     public function styleguideAction()
+        +     {
+        +         return $this->render('@ShopsysShop/Styleguide/styleguide.html.twig');
+        +     }
+        + }
+        ```
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: Changed

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 7]
```diff
@@ -639,6 +639,85 @@ There you can find links to upgrade notes for other versions too.
...
+    - add styleguide files according to PR
+        - [project-base/web/assets/styleguide/*](https://github.com/shopsys/shopsys/tree/master/project-base/web/assets/styleguide/)
+        - [project-base/src/Shopsys/ShopBundle/Resources/scripts/styleguide/*](https://github.com/shopsys/shopsys/tree/master/project-base/src/Shopsys/ShopBundle/Resources/scripts/styleguide/)
+        - [project-base/src/Shopsys/ShopBundle/Resources/views/Styleguide/*](https://github.com/shopsys/shopsys/tree/master/project-base/src/Shopsys/ShopBundle/Resources/views/Styleguide/)
+        - [project-base/src/Shopsys/ShopBundle/Resources/styles/styleguide/*](https://github.com/shopsys/shopsys/tree/master/project-base/src/Shopsys/ShopBundle/Resources/styles/styleguide/)
+
+    - update `gruntfile.js.twig` and add task
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: added link

File: upgrade/UPGRADE-v8.1.0-dev.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply upgrade/UPGRADE-v8.1.0-dev.md]: Yes that is what I mean :)

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 8]
```diff
@@ -639,6 +639,65 @@ There you can find links to upgrade notes for other versions too.
...
+        + ['admin/', 'common/', 'custom_admin/', 'frontend/', 'styleguide/']
+        ```
+
+    - update config file [.eslintignore](https://github.com/shopsys/shopsys/tree/master/project-base/.eslintignore)
+        ```diff
+          /src/Shopsys/ShopBundle/Resources/scripts/frontend/plugins
+        + /src/Shopsys/ShopBundle/Resources/scripts/styleguide
+        ```
+
+    - add route and contorller according to PR
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v8.1.0-dev.md]:
```suggestion
    - add route and controller according to PR
```

**Comment by TomasGottvald** [comment]
Removed unneeded framework notes, removed links, changed links to added files. Can you check it again please?

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 9]
```diff
@@ -639,6 +639,45 @@ There you can find links to upgrade notes for other versions too.
...
+
+    ```diff
+        + styleguide: {
+        +    files: {
+        +        'web/assets/styleguide/styles/styleguide_{{ cssVersion }}.css': '{{ customResourcesDirectory }}/styles/styleguide/main.less'
+        +    },
+        +    options: {
+        +        compress: true
+        +    }
+        + },
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v8.1.0-dev.md]: this is wrongly formatted

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 10]
```diff
@@ -639,6 +639,45 @@ There you can find links to upgrade notes for other versions too.
         ```
         - for more information you can see the [PR](https://github.com/shopsys/shopsys/pull/1461)
 
+- add Basic styleguide ([#1463](https://github.com/shopsys/shopsys/pull/1463))
+
+    - update config file [.eslintignore]
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v8.1.0-dev.md]:
```suggestion
    - update config file .eslintignore
```

File: upgrade/UPGRADE-v8.1.0-dev.md (hunk 8 above)
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: fixed

File: upgrade/UPGRADE-v8.1.0-dev.md (hunk 9 above)
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: fixed.

File: upgrade/UPGRADE-v8.1.0-dev.md (hunk 10 above)
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: hightlighted another way

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 11]
```diff
@@ -639,6 +639,45 @@ There you can find links to upgrade notes for other versions too.
...
         - for more information you can see the [PR](https://github.com/shopsys/shopsys/pull/1461)
 
+- add Basic styleguide ([#1463](https://github.com/shopsys/shopsys/pull/1463))
+
+    - update config file `.eslintignore`
+        ```diff
+          /src/Shopsys/ShopBundle/Resources/scripts/frontend/plugins
+        + /src/Shopsys/ShopBundle/Resources/scripts/styleguide
+        ```
+
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v8.1.0-dev.md]:
```suggestion
```

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 12]
```diff
@@ -639,6 +639,45 @@ There you can find links to upgrade notes for other versions too.
         ```
         - for more information you can see the [PR](https://github.com/shopsys/shopsys/pull/1461)
 
+- add Basic styleguide ([#1463](https://github.com/shopsys/shopsys/pull/1463))
+
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v8.1.0-dev.md]:
```suggestion
```

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 13]
```diff
@@ -639,6 +639,45 @@ There you can find links to upgrade notes for other versions too.
...
+        ```
+
+    - add route and controller according to PR
+        update `app/config/routing_dev.yml`
+        ```diff
+        + _styleguide:
+        +    path: /_styleguide/
+        +    defaults: { _controller: ShopsysShopBundle:Styleguide\Styleguide:styleguide}
+        ```
+
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v8.1.0-dev.md]:
```suggestion
```

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 14]
```diff
@@ -639,6 +639,45 @@ There you can find links to upgrade notes for other versions too.
...
+    - add styleguide controller:
+        - [src/Shopsys/ShopBundle/Controller/Styleguide/StyleguideController.php](https://github.com/shopsys/shopsys/blob/master/project-base/src/Shopsys/ShopBundle/Controller/Styleguide/StyleguideController.php)
+
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v8.1.0-dev.md]:
```suggestion
```

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 15]
```diff
@@ -639,6 +639,45 @@ There you can find links to upgrade notes for other versions too.
...
+    - add styleguide files according to PR
+
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v8.1.0-dev.md]:
```suggestion
```

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 16]
```diff
@@ -639,6 +639,45 @@ There you can find links to upgrade notes for other versions too.
...
+        - [web/assets/styleguide/*](https://github.com/shopsys/shopsys/tree/master/project-base/web/assets/styleguide/)
+        - [src/Shopsys/ShopBundle/Resources/scripts/styleguide/*](https://github.com/shopsys/shopsys/tree/master/project-base/src/Shopsys/ShopBundle/Resources/scripts/styleguide/)
+        - [src/Shopsys/ShopBundle/Resources/views/Styleguide/*](https://github.com/shopsys/shopsys/tree/master/project-base/src/Shopsys/ShopBundle/Resources/views/Styleguide/)
+        - [src/Shopsys/ShopBundle/Resources/styles/styleguide/*](https://github.com/shopsys/shopsys/tree/master/project-base/src/Shopsys/ShopBundle/Resources/styles/styleguide/)
+
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v8.1.0-dev.md]:
```suggestion
```

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 17]
```diff
@@ -639,6 +639,45 @@ There you can find links to upgrade notes for other versions too.
...
+    - update `src/Shopsys/ShopBundle/Resources/views/Grunt/gruntfile.js.twig` and add task
+
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v8.1.0-dev.md]:
```suggestion
```

**Pull request merged**